arithmetization/src/main/java/net/consensys/linea/zktracer/module/limits/L1Block.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.zktracer.module.limits;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.List;
+import java.util.Set;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.Accessors;
+import net.consensys.linea.zktracer.ColumnHeader;
+import net.consensys.linea.zktracer.module.Module;
+import net.consensys.linea.zktracer.module.hub.Hub;
+import net.consensys.linea.zktracer.opcode.OpCode;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Transaction;
+import org.hyperledger.besu.evm.log.Log;
+import org.hyperledger.besu.evm.log.LogTopic;
+import org.hyperledger.besu.evm.worldstate.WorldView;
+
+@RequiredArgsConstructor
+@Accessors(fluent = true)
+public class L1Block implements Module {
```

Comment:
Do you have a spec to compare the implementation to? I remember we had series of tickets, that I could find, but I wonder if there's a newer document where all of that compiled?

Reply:
[This PR](https://github.com/Consensys/zk-geth/pull/35) from Alex, which he didn't merge in the end.

Reply:
Also, this would deserve some equivalence testing. To make sure both implementations returns the same values. Seems easy enough and worthwhile

Reply:
Testing on “same values” is probably not a good idea as the Geth PR has neither been tested nor deployed.

If the prover team has some pairs of correct input/output, we would do our best to test them.

Reply:
can we add few unit test for the line counts using date you build.
Then we can add some tests using existing blocs from testnet based on our validation of this coming we.